quota.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package rate
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+// Quota tracks the remaining number of requests that can be made within a time
+// period.
+type Quota struct {
+	limit     *Limit
```

Comment:
do we need to use atomic.Pointer for the `limit` and `expiresAt` fields?

Reply:
Hmm, I think I might. I wonder if just having a RWMutex on a Quota would be better instead of using atomic pointers and atomic uint64.

Reply:
Refactored this to use a sync.RWMutex for the Quota, since all of the fields on a Quota should be guarded against concurrent access, with concurrent reads being fine.